README.md

```diff
@@ -1 +1,192 @@
+  it('should add two numbers', function () {
+    expect(addNumbers(1, 3)).toBe(4);
+  });
+});
+```
+
+What's this crazy syntax? When we run our tests, we use syntax that is provided to us by a framework called Jasmine. Jasmine gives us a set of predefined functions (such as `it`, `expect`, etc) that allow us to describe, quite simply, what we want the test to achieve.
+
+Above, we're telling our test runner that in our `Basic Functionality` block, we have a test that expects our number adding function to output `4` when we add `1` and `3` together.
+
+The test will then run, notice we haven't *actually* defined our number adding function, and fail. Right. Let's create the number adding function then!
+
+```js
+function addNumbers(numberOne, numberTwo) {
+}
+```
+
+Great - now we're getting started. Let's run our test again - doh! Failed. We haven't actually done any logic, so the function still won't output `4`, which is what our test is looking for.
+
+Simple - let's just return `4`! Then it passes all of our tests!
+
+```js
+function addNumbers(numberOne, numberTwo) {
+  return 4;
+}
+```
+
+Perfect, our tests now pass. Wait! Hold on a minute - when we try and use that function with any other numbers, we're going to always get `4` back! 
+
+How can we test for this? Test it again! Realistically, our tests should always test our functions with many different parameters, expecting correct answers from our functions. That way, if we have a test adding `1` and `3` (to equal `4`), and also adding `50` and `24329` (to equal `24379`), we can, with some confidence, know that our number adding function actually works as we expect.
+
+Let's modify our first test:
+
+```js
+describe('Basic Functionality', function () {
+  it('should add two numbers', function () {
+    expect(addNumbers(1, 3)).toBe(4);
+    expect(addNumbers(343, 9283)).toBe(9626);
+    expect(addNumbers(1223, 21)).toBe(1244);
+    expect(addNumbers(10, 653)).toBe(663);
+  });
+});
+```
+
+Okay, now we've got multiple test cases, let's run our tests again. As expected, our first one passed (as our number adding function is only returning `4`), but the others have failed. Now we need to head back to our number adder and actually implement it correctly.
+
+```js
+function addNumbers(numberOne, numberTwo) {
+  return numberOne + numberTwo;
+}
+```
+
+We run our tests again and they all pass - happy days! The code goes into production, but two days later we receive a bug report saying that when people are adding `2` and `2` together, they get `22` instead of `4`. Oh no, what could've gone wrong?
+
+Our function doesn't support strings! In JavaScript, if we were to do `"2" + 2`, we'd get `22` as the result because the `+` operator adds two strings together, as well as actually doing addition between two numbers.
+
+Let's head back to our tests, and add some use-cases for when the function gets called with parameters that don't match what it expects.
```

Comment:
really nice example - this is a great demonstration of why it's important to have good tests